kube-core/src/metadata.rs

```diff
@@ -23,39 +23,57 @@ pub struct TypeMeta {
 /// schema without knowing the details of the version.
 #[derive(Deserialize, Serialize, Clone, Default, Debug)]
 #[serde(rename_all = "camelCase")]
-pub struct PartialObjectMeta {
+pub struct PartialObjectMeta<K> {
     /// The type fields, not always present
     #[serde(flatten, default)]
     pub types: Option<TypeMeta>,
     /// Standard object's metadata
     #[serde(default)]
     pub metadata: ObjectMeta,
+    /// Type information for static dispatch
+    #[serde(skip, default)]
+    pub _phantom: PhantomData<K>,
 }
 
-impl From<PartialObjectMeta> for ObjectMeta {
-    fn from(obj: PartialObjectMeta) -> Self {
+// Users usually want the inner metadata on returns
+impl<K> From<PartialObjectMeta<K>> for ObjectMeta {
+    fn from(obj: PartialObjectMeta<K>) -> Self {
         ObjectMeta { ..obj.metadata }
```

Comment:
Isn't this equivalent to:

```suggestion
        obj.metadata
```

though honestly I'm not quite sure this impl saves much work for users...

Reply:
Arguably it's too late to remove this anyway since it got out in 0.79. Oh well... :P

Reply:
Oh, this is a mistake to leave in for sure. It has worse ergonomics than using either `.metadata` or the `Resource` impl  on `PartialObjectMeta<K>` now. I will take it out. I think the correct solution for stability would have been to release the feature under an unstable flag, but this is a relatively minor breakage on a new feature.